### PR TITLE
Python Viewer Navmesh Visualization via Key Press Toggle

### PR DESCRIPTION
**Pull request by JuanTheEngineer**
## Motivation and Context

This PR migrates the Navmesh Recompute and Toggle feature to the python viewer.

## How Has This Been Tested

Building and running `viewer.py`

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: examples/viewer.py [hunk 1]
```diff
@@ -80,6 +83,9 @@ def __init__(self, sim_settings: Dict[str, Any]) -> None:
         self.sim: habitat_sim.simulator.Simulator = None
         self.reconfigure_sim()
 
+        # compute NavMesh
+        self.navmesh_config_and_recompute()
```
Comment (aclegg3) [inline examples/viewer.py:87]: As @Skylion007 suggested:
```suggestion
        # compute NavMesh if not already loaded by the scene.
        if not self.sim.pathfinder.is_loaded:
            self.navmesh_config_and_recompute()
```

File: examples/viewer.py [hunk 2]
```diff
@@ -193,6 +199,7 @@ def reconfigure_sim(self) -> None:
         self.active_scene_graph = self.sim.get_active_scene_graph()
         self.default_agent = self.sim.get_agent(self.agent_id)
         self.agent_body_node = self.default_agent.scene_node
+        self.agent_body_node.translation = [2.65, 0.5, 4.5]
```
Comment (aclegg3) [inline examples/viewer.py:202]: Let's not hardcode location in viewer.

**Review by aclegg3 — APPROVED** [review]
Small fixes and ready to go.

File: examples/viewer.py [hunk 3]
```diff
@@ -274,6 +282,17 @@ def key_press_event(self, event: Application.KeyEvent) -> None:
             self.invert_gravity()
             logger.info("Command: gravity inverted")
 
+        elif key == pressed.N:
+            if event.modifiers == mod.SHIFT:
+                logger.info("Command: recompute navmesh")
+                self.navmesh_config_and_recompute()
+            else:
+                if self.navmesh_success:
```
Comment (Skylion007) [inline examples/viewer.py:290]:
```suggestion
                if self.navmesh_success or self.sim.pathfinder.is_loaded:
```

File: examples/viewer.py (hunk 1 above)
Comment (Skylion007) [reply examples/viewer.py:87]: You also need to set `self.navmesh_success ` to `self.sim.pathfinder.is_loaded`. In fact, I would replace that variable with a direct access to `self.sim.pathfinder.is_loaded` and only use the `recompute_navmesh` return value to throw an error / exception if it fails.

Comment (aclegg3) [reply examples/viewer.py:87]: Yeah, agreed. Let's just use `is_loaded` everywhere instead of tracking the recompute with a variable.

**Pull request merged**